### PR TITLE
Cas Airplane

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

3Am
added to all ships but sucklaco because no space
## Why It's Good For The Game

content
## Changelog
:cl:
add: New CAS airplane for POs
add: The CAS jet needs to be fueled using jerrycans
add: Click while in CAS mode to attack ground targets
add: Weapons can be attached to the plane like normal cas but must be selected for firing
imageadd: Sprites by darkplasmaray
code: cleaned up some UIs, dropship code and CAS weapons code
/:cl:

### CODE REVIEW

File: code/modules/shuttle/cas_plane.dm [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+		return
+	switch(owner.state)
+		if(PLANE_STATE_DEACTIVATED)
+			set_cockpit_overlay("cockpit_opening")//flick doesnt work here, thanks byond
+			sleep(7)
+			set_cockpit_overlay("cockpit_open")
+			owner.state = PLANE_STATE_ACTIVATED
+			return
+		if(PLANE_STATE_PREPARED | PLANE_STATE_FLYING)
+			to_chat("<span class='warning'>The plane is in-flight!!</span>")
```
Comment (XSlayer300) [inline code/modules/shuttle/cas_plane.dm:66]:
```suggestion
			to_chat("<span class='warning'>The plane is in-flight!</span>")
```

File: code/modules/shuttle/cas_plane.dm [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+			return
+		if(PLANE_STATE_ACTIVATED)
+			if(occupant)
+				to_chat(user, "<span class='warning'>Someones already inside!</span>")
```
Comment (XSlayer300) [inline code/modules/shuttle/cas_plane.dm:70]:
```suggestion
				to_chat(user, "<span class='warning'>Someone is already inside!</span>")
```

File: code/modules/shuttle/cas_plane.dm [hunk 3]
```diff
@@ -0,0 +1,411 @@
...
+				return
+			to_chat(user, "<span class='notice'>Youu start climbing into the cockpit...</span>")
```
Comment (XSlayer300) [inline code/modules/shuttle/cas_plane.dm:72]:
```suggestion
			to_chat(user, "<span class='notice'>You start climbing into the cockpit...</span>")
```

File: code/modules/shuttle/cas_plane.dm [hunk 4]
```diff
@@ -0,0 +1,411 @@
...
+
+/obj/structure/caspart/caschair/connect_to_shuttle(obj/docking_port/mobile/port, obj/docking_port/stationary/dock, idnum, override=FALSE)
+	if(!istype(port, /obj/docking_port/mobile/marine_dropship/casplane))
+		return
+	var/obj/docking_port/mobile/marine_dropship/casplane/plane = port
+	owner = plane
+	plane.chair = src
+
+/obj/docking_port/stationary/marine_dropship/cas
+	name = "CAS plane Hangar Pad"
```
Comment (XSlayer300) [inline code/modules/shuttle/cas_plane.dm:134]:
```suggestion
	name = "CAS plane hangar pad"
```

File: code/modules/shuttle/cas_plane.dm [hunk 5]
```diff
@@ -0,0 +1,411 @@
...
+			else
+				state = PLANE_STATE_ACTIVATED
+
+///Runs checks and creates a new eye/hands over control to the eye
+/obj/docking_port/mobile/marine_dropship/casplane/proc/begin_cas_mission(mob/living/user)
+	if(!fuel_left)
+		to_chat(user, "<span class='warning'>No fuel remaining!</span>")
+		return
+	if(state != PLANE_STATE_FLYING)
+		to_chat(user, "<span class='warning'>You're in-flight!</span>")
```
Comment (XSlayer300) [inline code/modules/shuttle/cas_plane.dm:241]:
```suggestion
		to_chat(user, "<span class='warning'>You are in-flight!</span>")
```

**Comment by neohenry** [comment]
IT'S HAPPENING!

IT'S HAPPENING!

**Comment by sunofang** [comment]
mispeling tiaem.

**Comment by Pariah919** [comment]
Flying this makes the FOB drone no longer function, which can possibly cause some issues if a PO decides to take a joyride before anyone uses it.

**Comment by TiviPlus** [comment]
feature

**Comment by XSlayer300** [comment]
Add an intercomms inside the condor jet, so I would know if I committed FF or not.

Also, make landings always available regardless of fuel so I won't get stuck when I ran out somehow.

**Comment by TiviPlus** [comment]
Landing should happen automatically when you run out of fuel
Intercomms was a bug, fixed

**Comment by David-Stormwell** [comment]
Begin Firemission option doesn't give any options for targets.
It should give a list of targets, as CAS flares exists and I feel you should have the option of selecting one.

**Comment by Pariah919** [comment]
This should be untmed until you can no longer sit in fire mission mode infinitely, allowing you to snipe benos even after the dot is gone, or make the view range based on the dot rather than AI eye from marines.

**Comment by TiviPlus** [comment]
Already fixed

**Comment by Hughgent** [comment]
Someone is complaining that they cannot reload the nose gun.

**Comment by TiviPlus** [comment]
also should be fixed, tell keyholders to update tms

**Comment by XSlayer300** [comment]
You can abuse the use of the CAS plane fuel by launching and then disabling the engines as it is launching, it will not consume any fuel.

And somehow I died like this.
image

**Comment by TiviPlus** [comment]
Spurious mapdiff fail

**Comment by Denthamos** [comment]
Bug I noticed with the CAS craft.  When a synth was doing lased targeting for me with it, it was giving me vision across the entire map with no static.  When a normal marine lased the static was there.  Only thing I noticed different between the mobs was synth was wearing mesons.

Secondly, dear god this thing eats up fuel like crazy.  In a single hour it went through three fuel tanks as it empties pretty damn fast, each refilling taking a jerry can and a third to fill completely which is about 225-250 fuel.  Fuel tank holds 1200 fuel.  Might need to consider either reducing the fuel cost or adding an extra fuel tank or two in the hangar armory.

File: code/game/objects/structures/dropship_equipment.dm [hunk 6]
```diff
@@ -132,120 +144,132 @@
...
-	var/equip_category //on what kind of base this can be installed.
-	var/obj/effect/attach_point/ship_base //the ship base the equipment is currently installed on.
-	var/uses_ammo = FALSE //whether it uses ammo
-	var/obj/structure/ship_ammo/ammo_equipped //the ammo currently equipped.
-	var/is_weapon = FALSE //whether the equipment is a weapon usable for dropship bombardment.
-	var/obj/machinery/computer/dropship_weapons/linked_console //the weapons console of the dropship we're installed on.
-	var/is_interactable = FALSE //whether they get a button when shown on the shuttle console's equipment list.
+	///on what kind of base this can be installed.
+	var/equip_category
+	//the ship base the equipment is currently installed on.
```
Comment (Hughgent) [inline code/game/objects/structures/dropship_equipment.dm:149]: three /// or no?

**Review by Hughgent — APPROVED** [review]

File: code/game/objects/structures/dropship_equipment.dm (hunk 6 above)
Comment (TiviPlus) [inline code/game/objects/structures/dropship_equipment.dm:149]:
```suggestion
	var/equip_category
	///the ship base the equipment is currently installed on.
```

**Review by Hughgent — APPROVED** [review]

**Pull request merged**